openmc/mgxs_library.py

```diff
@@ -333,37 +376,51 @@ def alias(self, alias):
         else:
             self._alias = self._name
 
+    @zaid.setter
+    def zaid(self, zaid):
+        # Check type and value
+        check_type('zaid', zaid, Integral)
+        check_greater_than('zaid', zaid, 0, equality=False)
+        self._zaid = zaid
+
+    @awr.setter
+    def awr(self, awr):
+        # Check validity of type and that the awr value is > 0
+        check_type('awr', awr, Real)
+        check_greater_than('awr', awr, 0.0, equality=False)
+        self._awr = awr
+
     @kT.setter
     def kT(self, kT):
         # Check validity of type and that the kT value is >= 0
-        check_type("kT", kT, Real)
-        check_greater_than("kT", kT, 0.0, equality=True)
+        check_type('kT', kT, Real)
+        check_greater_than('kT', kT, 0.0, equality=True)
         self._kT = kT
 
     @scatt_type.setter
     def scatt_type(self, scatt_type):
         # check to see it is of a valid type and value
-        check_value("scatt_type", scatt_type, ['legendre', 'histogram',
+        check_value('scatt_type', scatt_type, ['legendre', 'histogram',
                                                'tabular'])
         self._scatt_type = scatt_type
 
     @order.setter
     def order(self, order):
         # Check type and value
-        check_type("order", order, Integral)
-        check_greater_than("order", order, 0, equality=True)
+        check_type('order', order, Integral)
+        check_greater_than('order', order, 0, equality=True)
         self._order = order
 
     @tabular_legendre.setter
     def tabular_legendre(self, tabular_legendre):
         # Check to make sure this is a dict and it has our keys with the
         # right values.
-        check_type("tabular_legendre", tabular_legendre, dict)
+        check_type('tabular_legendre', tabular_legendre, dict)
         if 'enable' in tabular_legendre:
             enable = tabular_legendre['enable']
             check_type('enable', enable, bool)
         else:
-            msg = "enable must be provided in tabular_legendre"
+            msg = 'enable must be provided in tabular_legendre'
```

Comment:
I think this could be a bit confusing to a novice - perhaps this?

```python
msg = 'The tabular_legendre dict must include a value keyed by "enable"'
```